dist/immstruct.js

```diff
@@ -653,13 +824,12 @@ function Structure (options) {
     this._currentRevision = 0;
   }
 
-  this._pathListeners = [];
+  this._pathListeners = {};
```

Comment:
all the callbacks get stored in a tree structure that matches the path. That tree structure is traversed up from the path of the change event to collect all the callbacks associated with each level of the tree. I believe this is the same as what the prefix check was accomplishing.